azure-iot-hub-devicesdk/samples/receive_c2d_message.py

```diff
@@ -18,26 +19,27 @@
 device_client = DeviceClient.from_authentication_provider(auth_provider, "mqtt")
 
 
-# The connection state callback allows us to detect when the client is connected and disconnected:
-def connection_state_callback(status):
-    print("connection status: " + status)
+# connect the client.
+device_client.connect()
 
+# enable the device to receive c2d messages
+c2d_message_queue = device_client.get_c2d_message_queue()
 
-def c2d_message_handler(c2d_message):
-    print("the data in the message received was ")
-    print(c2d_message.data)
-    print("custom properties are")
-    print(c2d_message.custom_properties)
 
+# define behavior for receiving a C2D message
+def c2d_listener(message_queue):
+    while True:
+        c2d_message = message_queue.get()  # blocking call
+        print("the data in the message received was ")
```

Comment:
I guess we could also use get_nowait() and be non-blocking right?

Reply:
That is correct

---
In reply to: [258282020](https://github.com/Azure/azure-iot-sdk-python-preview/pull/49#discussion_r258282020) [](ancestors = 258282020)